### PR TITLE
Defines "-- rollback empty" as the way to specify "no rollback needed". 

### LINKED ISSUE #3069: How to specify an Empty Rollback using SQL Dialect
## Environment

**Liquibase Version**: 4.13
**Liquibase Integration & Version**: GitHub Action
**Liquibase Extension(s) & Version**: 7
**Database Vendor & Version**: Snowflake
**Operating System Type & Version**: Linux
**Infrastructure Type/Provider**: Azure

## Description

Unable to specify an empty rollback for a changeset using the SQL dialect.

## Steps To Reproduce

Using a SQL dialect, consider a change set like:
```sql
-- liquibase formatted sql

-- changeset anthonyr:create-table-AW_TEST-1 runOnChange:true
create or replace table AW_TEST (
  ID int not null autoincrement,
  valueDate date not null
)
-- rollback
```
When I test a rollback, Liquibase throws an error:
```
Unexpected error running Liquibase: No inverse to liquibase.change.core.RawSQLChange created
```

I am legitimately OK with an [empty](https://docs.liquibase.com/change-types/empty.html) rollback.  Note that I don't really see a SQL Example in the documentation.

## Actual Behavior

When I test a rollback, Liquibase throws an error:
```
Unexpected error running Liquibase: No inverse to liquibase.change.core.RawSQLChange created
```

## Expected/Desired Behavior

When I test a rollback, Liquibase should basically do nothing for this changeset since the rollback is empty.

### PR DESCRIPTION
**Pull request by filipelautert**
## Impact
- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description
- Defines "-- rollback empty" as the way to specify "no rollback needed". 
- Adds detailed message output in cases it may have been missed.
- Fixes #3069 

## Things to worry about
Documentation needs to be updated to add the new magic word:
- https://docs.liquibase.com/commands/rollback/rollback-by-tag.html
- https://docs.liquibase.com/workflows/liquibase-community/liquibase-auto-rollback.html

I found those 2 pages, but any other pages having the expression "<rollback />" may be a candidate.

## Additional Context

Documentation update to reflect this new feature: https://github.com/Datical/liquibase-docs/pull/164

### CODE REVIEW

**Review by nvoxland — APPROVED** [review]
Code review:

Things to be aware of:
- Adds "empty" as a keyword meaning "no rollback". It is checked case insensitively
- Added an integration test for rollback in general for formatted sql
- Test harness failures are unrelated

Things to worry about:
- Nothing

File: liquibase-core/src/main/java/liquibase/changelog/ChangeSet.java [hunk 1]
```diff
@@ -827,6 +827,10 @@ public void rollback(Database database, ChangeExecListener listener) throws Roll
                 List<Change> changes = getChanges();
                 for (int i = changes.size() - 1; i >= 0; i--) {
                     Change change = changes.get(i);
+                    if (change instanceof RawSQLChange) {
+                        throw new RollbackFailedException("Liquibase does not support automatic rollback generation for raw " +
+                            "sql changes (did you mean to specify keyword \"empty\" to ignore rolling back this change?)");
```
Comment (XDelphiGrl) [inline liquibase-core/src/main/java/liquibase/changelog/ChangeSet.java:832]: This error message! Is great!

**Review by suryaaki2 — APPROVED** [review]

File: liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java [hunk 2]
```diff
@@ -283,6 +286,26 @@ private Liquibase createLiquibase(String changeLogFile, ResourceAccessor resourc
...
+        Liquibase liquibase = createLiquibase(emptyRollbackSqlChangeLog);
+        clearDatabase();
+
+        liquibase = createLiquibase(emptyRollbackSqlChangeLog);
+        liquibase.update(this.contexts);
+
+        liquibase = createLiquibase(emptyRollbackSqlChangeLog);
+        liquibase.rollback(new Date(0), this.contexts);
+
+        liquibase = createLiquibase(emptyRollbackSqlChangeLog);
```
Comment (XDelphiGrl) [inline liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java:302]: The update and rollback at line 302 and line 305 are duplicates of the update and rollback at lines 299 and 302. Did you mean to use a different changelog?

**Review by XDelphiGrl — CHANGES_REQUESTED** [review]
Hello, @filipelautert! Can you please take a look at my comment in the test class? If I am misunderstanding something, let me know; otherwise, please remove the duplication. The rest of the code looks good. 

CHANGES REQUESTED

File: liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java (hunk 2 above)
Comment (filipelautert) [reply liquibase-integration-tests/src/test/java/liquibase/dbtest/AbstractIntegrationTest.java:302]: No, sorry. It is duplicated code, I'm removing it. thanks for spot that.

**Review by XDelphiGrl — APPROVED** [review]
There are instances where you may not need to define a rollback for a specific change in a formatted SQL changelog. However, without defining one, commands such as `rollback-count` fail due to missing rollback SQL for a raw SQL change. This PR allows users to specify changesets that do not require a rollback statement by adding `--rollback empty` to the changeset. 
* New integration test added.
* No additional testing necessary.

APPROVED

**Pull request merged**